presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -106,7 +107,8 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
                     semiJoinSymbol,
                     Optional.empty(),
                     Optional.empty(),
-                    Optional.empty()
+                    Optional.empty(),
+                    SystemSessionProperties.isLegacySemiJoin(context.get())
```

Comment:
Since importance of ternary logic or binary logic could be deduced before `SemiJoinNode` creation (and without looking into session at all) I think it's good approach to store the flag in `SemiJoinNode`